working_villagers/api.lua

```diff
@@ -320,6 +366,305 @@ function working_villages.villager:move_main_to_wield(pred)
+  return self.object:add_velocity(vel)
+end
+
+function working_villages.villager:move_to(...)
+  return self.object:move_to(...)
+end
+function working_villages.villager:punch(...)
+  return self.object:punch(...)
+end
+function working_villages.villager:right_click(clicker)
+  return self.object:right_click(clicker)
+end
+
+function working_villages.villager:get_wield_list()
+  return self.object:get_wield_list()
+end
+
+function working_villages.villager:get_armor_groups()
+  return self.object:get_armor_groups()
+end
+function working_villages.villager:set_armor_groups(groups)
+  return self.object:set_armor_groups(groups)
+end
+
+function working_villages.villager:get_animation()
+  return self.object:get_animation()
+end
+
+function working_villages.villager:set_animation_frame_speed(frame_speed)
+  return self.object:set_animation_frame_speed(frame_speed)
+end
+function working_villages.villager:get_attach()
+  return self.object:get_attach()
+end
+function working_villages.villager:set_attach(...)
+  return self.object:set_attach(...)
+end
+function working_villages.villager:get_children()
+  return self.object:get_children()
+end
+function working_villages.villager:set_detach()
+  return self.object:set_detach()
+end
+
+function working_villages.villager:get_bone_position()
+  return self.object:get_bone_position()
+end
+function working_villages.villager:set_bone_position(...)
+  return self.object:set_bone_position(...)
+end
+
+function working_villages.villager:set_properties(vel)
+  return self.object:set_properties(vel)
+end
+
+function working_villages.villager:get_nametag_attributes()
+  return self.object:get_nametag_attributes()
+end
+function working_villages.villager:set_nametag_attributes(vel)
+  return self.object:set_nametag_attributes(vel)
+end
+
+-- lua entity
+function working_villages.villager:remove()
+  return self.object:remove()
+end
+
+function working_villages.villager:set_velocity(vel)
+  return self.object:set_velocity(vel)
+end
+
+function working_villages.villager:get_acceleration()
+  return self.object:get_acceleration()
+end
+function working_villages.villager:set_acceleration(acc)
+  return self.object:set_acceleration(acc)
+end
+
+function working_villages.villager:get_rotation()
+  return self.object:get_rotation()
+end
+function working_villages.villager:set_rotation(rot)
+  return self.object:set_rotation(rot)
+end
+
+function working_villages.villager:get_yaw()
+  return self.object:get_yaw()
+end
+function working_villages.villager:set_yaw(yaw)
+  return self.object:set_yaw(yaw)
+end
+
+function working_villages.villager:get_texture_mod()
+  return self.object:get_texture_mod()
+end
+function working_villages.villager:set_texture_mod(mod)
+  return self.object:set_texture_mod(mod)
+end
+
+function working_villages.villager:set_sprite(...)
+  return self.object:set_sprite(...)
+end
+
+function working_villages.villager:get_luaentity()
+  return self.object:get_luaentity()
+end
+
+-- player specific
```

Comment:
I only needed a few of them for the `gardener` and `wizard`. The rest are included for completeness (i.e., to support other mods which I am not testing). I moved them to a separate file, since they're boilerplate, numerous and I copied them from another mod anyway.